packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -202,40 +384,103 @@ class Actions extends InheritedWidget {
   /// passed in here (e.g. a final variable from your widget class) instead of
   /// defining it inline in the build function.
   /// {@endtemplate}
-  final Map<LocalKey, ActionFactory> actions;
+  final Map<Type, Action<Intent>> actions;
+
+  /// {@macro flutter.widgets.child}
+  final Widget child;
+
+  // Visits the Actions widget ancestors of the given element using
+  // getElementForInheritedWidgetOfExactType. Returns true if the visitor found
+  // what it was looking for.
+  static bool _visitActionsAncestors(BuildContext context, bool visitor(InheritedElement element)) {
+    InheritedElement actionsElement = context.getElementForInheritedWidgetOfExactType<_ActionsMarker>();
+    while (actionsElement != null) {
+      if (visitor(actionsElement) == true) {
+        break;
+      }
+      // _getParent is needed here because
+      // context.getElementForInheritedWidgetOfExactType will return itself if it
+      // happens to be of the correct type.
+      final BuildContext parent = _getParent(actionsElement);
+      actionsElement = parent.getElementForInheritedWidgetOfExactType<_ActionsMarker>();
+    }
+    return actionsElement != null;
+  }
 
   // Finds the nearest valid ActionDispatcher, or creates a new one if it
   // doesn't find one.
-  static ActionDispatcher _findDispatcher(Element element) {
-    assert(element.widget is Actions);
-    final Actions actions = element.widget as Actions;
-    ActionDispatcher dispatcher = actions.dispatcher;
-    if (dispatcher == null) {
-      bool visitAncestorElement(Element visitedElement) {
-        if (visitedElement.widget is! Actions) {
-          // Continue visiting.
-          return true;
-        }
-        final Actions actions = visitedElement.widget as Actions;
-        if (actions.dispatcher == null) {
-          // Continue visiting.
-          return true;
-        }
-        dispatcher = actions.dispatcher;
-        // Stop visiting.
-        return false;
+  static ActionDispatcher _findDispatcher(BuildContext context) {
+    ActionDispatcher dispatcher;
+    _visitActionsAncestors(context, (InheritedElement element) {
+      final ActionDispatcher found = (element.widget as _ActionsMarker).dispatcher;
+      if (found != null) {
+        dispatcher = found;
+        return true;
       }
+      return false;
+    });
+    return dispatcher ?? const ActionDispatcher();
+  }
 
-      element.visitAncestorElements(visitAncestorElement);
+  /// Returns a [VoidCallback] handler that invokes the bound action for the
+  /// given intent if the action is enabled, and returns null if the action is
+  /// not enabled.
+  ///
+  /// This is intended to be used in widgets which have something similar to an
+  /// `onTap` handler, which takes a `VoidCallback`, and can be set to the
+  /// result of calling this function.
+  static VoidCallback handler<T extends Intent>(BuildContext context, T intent, {bool nullOk = false}) {
```

Comment:
My understanding is that in order to use this correctly, you're not allowed to cache the returned handler between builds. When the context's dependencies change, you are required to call this again in case the action is disabled now. Maybe this should be part of the documentation?

Reply:
e.g. somehow mention that this creates a dependency similarly to how it's explained on the find method.